pilot/pkg/proxy/envoy/v2/lds_test.go

```diff
@@ -396,6 +398,94 @@ func TestLDS(t *testing.T) {
 	// TODO: dynamic checks ( see EDS )
 }
 
+// TestLDS using default sidecar in root namespace
+func TestLDSWithSidecarForWorkloadWithoutService(t *testing.T) {
+	server, tearDown := util.EnsureTestServer(func(args *bootstrap.PilotArgs) {
+		args.Plugins = bootstrap.DefaultPlugins
+		args.Config.FileDir = env.IstioSrc + "/tests/testdata/networking/sidecar-without-service"
+		args.Mesh.MixerAddress = ""
+		args.Mesh.RdsRefreshDelay = nil
+		args.Mesh.ConfigFile = env.IstioSrc + "/tests/testdata/networking/sidecar-without-service/mesh.yaml"
+		args.Service.Registries = []string{}
+	})
+	registry := memServiceDiscovery(server, t)
+	registry.AddWorkload("98.1.1.1", model.Labels{"app": "consumeronly"}) // These labels must match the sidecars workload selector
+
+	testEnv = testenv.NewTestSetup(testenv.SidecarConsumerOnlyTest, t)
+	testEnv.Ports().PilotGrpcPort = uint16(util.MockPilotGrpcPort)
+	testEnv.Ports().PilotHTTPPort = uint16(util.MockPilotHTTPPort)
+	testEnv.IstioSrc = env.IstioSrc
+	testEnv.IstioOut = env.IstioOut
+
+	server.EnvoyXdsServer.ConfigUpdate(true)
+	defer tearDown()
+
+	adsResponse, err := adsc.Dial(util.MockPilotGrpcAddr, "", &adsc.Config{
+		Meta: map[string]string{
+			model.NodeMetadataConfigNamespace:   "consumerns",
+			model.NodeMetadataInstanceIPs:       "98.1.1.1", // as service instance of ingress gateway
+			model.NodeMetadataIstioProxyVersion: "1.1.0",
+		},
+		IP:        "98.1.1.1",
+		Namespace: "consumerns", // namespace must match the namespace of the sidecar in the configs.yaml
+		NodeType:  "sidecar",
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer adsResponse.Close()
+
+	adsResponse.DumpCfg = true
+	adsResponse.Watch()
+
+	_, err = adsResponse.Wait("lds", 10000*time.Second)
+	if err != nil {
+		t.Fatal("Failed to receive LDS response", err)
+		return
+	}
+	_, err = adsResponse.Wait("lds", 10000*time.Second)
```

Comment:
I realize the other tests do the same, but we probably shouldn't wait for 3 hours